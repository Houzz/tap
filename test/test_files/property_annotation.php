<?php

class A {
    public function fa() {}
}

class B {
    /** @var A */
    private $a;

    public function fb() {
        $aa = new A();
        $aa->fa();
        $aa->fb(); // error

        $this->a = new A();
        $this->a->fa();
        $this->a->fb(); // error

        $this->a = new C(); // error
        $this->a = new D();
    }
}

class C {}
class D extends A {}

class A2 {
    /** @var A */
    public $a, $b = null, $c = 2;

    /** @var int */
    public $inta = 3, $intb;

    public $noDocA = "str", $noDocB;

    public function f() {
        __tap_check($this->a, "A");
        __tap_check($this->b, "A|null");
        __tap_check($this->c, "A");
        __tap_check($this->inta, "int");
        __tap_check($this->intb, "int");
        __tap_check($this->noDocA, "string");
        __tap_check($this->noDocB, "any");
    }
}

