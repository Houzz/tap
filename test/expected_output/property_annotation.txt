14 E_METHOD_NOT_FOUND
18 E_METHOD_NOT_FOUND
20 E_EXPECTED_TYPE
30 E_TYPE_HINT
